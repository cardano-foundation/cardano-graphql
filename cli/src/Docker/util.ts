import { QuestionCollection } from 'inquirer'
import { DockerComposeStack, StatefulService } from './index'

export function selectServicesFromStack (stack: DockerComposeStack, multi = true): QuestionCollection {
  const label = multi === true ? 'services' : 'service'
  return {
    name: label,
    message: `Select ${label}`,
    type: multi === true ? 'checkbox' : 'list',
    choices: [...stack.statefulServices.values()].map(service => {
      return {
        name: service.name,
        value: service
      }
    }),
    validate (input) {
      if (input.length === 0) {
        throw new Error('At least one service must be selected')
      }
      return true
    }
  }
}

export function servicesFromNameArg (stack: DockerComposeStack, names: string): StatefulService[] {
  return names.split(',').map((name: string) => {
    if (!stack.statefulServices.has(name)) {
      throw new Error(`Unknown service ${name}`)
    }
    return stack.statefulServices.get(name)
  })
}
