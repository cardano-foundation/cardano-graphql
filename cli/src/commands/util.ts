import { DockerComposeStack } from '../docker'

export function serviceChoices (stack: DockerComposeStack) {
  return [...stack.statefulServices.values()].map(service => {
    return { name: service.name, value: service }
  })
}

export function serviceFromName (stack: DockerComposeStack, name: string) {
  if (!stack.statefulServices.has(name)) {
    throw new Error(`Unknown service ${name}`)
  }
  return stack.statefulServices.get(name)
}

export function servicesFromNames (stack: DockerComposeStack, names: string) {
  return names.split(',').map((name: string) => serviceFromName(stack, name))
}
