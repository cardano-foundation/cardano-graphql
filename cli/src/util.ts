
export function ensureValue (input: string) {
  return !input ? 'Cannot be blank' : true
}
