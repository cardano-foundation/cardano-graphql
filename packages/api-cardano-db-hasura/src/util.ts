import { ConnectionConfig, createInteractionContext } from '@cardano-ogmios/client'
import { Logger } from 'ts-log'

export const createInteractionContextWithLogger = (connection: ConnectionConfig, logger: Logger, module: string, onClose?: () => Promise<void>) =>
  createInteractionContext(
    (error) => {
      logger.error({ module, error }, error.message)
    },
    async (code, reason) => {
      if (code === 1000) {
        logger.info({ module, code }, reason)
      } else {
        logger.error({ module, code }, 'Connection closed')
        await onClose?.()
      }
    },
    {
      connection
    }
  )
