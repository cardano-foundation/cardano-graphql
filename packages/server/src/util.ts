import { Application } from 'express'
import http from 'http'

export function listenPromise (app: Application, port: number, hostname?: string): Promise<http.Server> {
  return new Promise(function (resolve, reject) {
    const server: http.Server = app.listen(port, hostname, () => resolve(server))
    server.on('error', reject)
  })
}
