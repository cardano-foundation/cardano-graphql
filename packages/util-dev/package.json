{
  "name": "@cardano-graphql/util-dev",
  "version": "1.0.0",
  "description": "Common development utilities",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "scripts": {
    "build": "tsc -b ./src",
    "cleanup": "shx rm -rf dist node_modules",
    "lint": "eslint --ignore-path ../../.eslintignore \"**/*.ts\"",
    "prepack": "yarn build",
    "test": "shx echo No tests in this package"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/input-output-hk/cardano-graphql.git"
  },
  "author": "Rhys Bartels-Waller",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/input-output-hk/cardano-graphql/issues"
  },
  "homepage": "https://github.com/input-output-hk/cardano-graphql/blob/master/packages/util-dev/README.md",
  "devDependencies": {
    "@cardano-graphql/util": "1.0.0",
    "shx": "^0.3.2"
  },
  "directories": {
    "lib": "src",
    "test": "test"
  },
  "files": [
    "dist",
    "LICENSE",
    "NOTICE",
    "README.md"
  ]
}
