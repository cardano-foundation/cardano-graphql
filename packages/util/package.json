{
  "name": "@cardano-graphql/util",
  "version": "1.0.0",
  "description": "Common utilities",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "scripts": {
    "build": "tsc -b ./src",
    "cleanup": "shx rm -rf dist node_modules",
    "lint": "eslint --ignore-path ../../.eslintignore \"**/*.ts\"",
    "prepack": "yarn build",
    "test": "shx echo No tests in this package"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/input-output-hk/cardano-graphql.git"
  },
  "author": "Rhys Bartels-Waller",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/input-output-hk/cardano-graphql/issues"
  },
  "homepage": "https://github.com/input-output-hk/cardano-graphql/blob/master/packages/util/README.md",
  "dependencies": {
    "graphql": "14.5.8",
    "graphql-bigint": "^1.0.0",
    "graphql-tag": "^2.10.3",
    "p-retry": "^4.2.0"
  },
  "devDependencies": {
    "@types/node": "^14.0.13",
    "shx": "^0.3.2",
    "typescript": "^3.9.5"
  },
  "directories": {
    "lib": "src",
    "test": "test"
  },
  "files": [
    "dist",
    "LICENSE",
    "NOTICE",
    "README.md"
  ]
}
