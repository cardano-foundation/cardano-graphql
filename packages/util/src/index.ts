import { onFailedAttemptFor } from './onFailedAttemptFor'
import { loadQueryNode } from './queryNodeLoading'
import * as dateTimeValidation from './dateTimeValidation'
import * as scalars from './scalars'

export default {
  dateTimeValidation,
  onFailedAttemptFor,
  loadQueryNode,
  scalars
}
