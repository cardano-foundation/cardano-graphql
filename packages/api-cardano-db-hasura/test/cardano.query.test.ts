import path from 'path'

import { DocumentNode } from 'graphql'
import util from '@cardano-graphql/util'
import { TestClient } from '@cardano-graphql/util-dev'
import { buildClient } from './util'

function loadQueryNode (name: string): Promise<DocumentNode> {
  return util.loadQueryNode(path.resolve(__dirname, '..', 'src', 'example_queries', 'cardano'), name)
}

describe('cardano', () => {
  let client: TestClient
  beforeAll(async () => {
    client = await buildClient()
  }, 60000)

  it('Returns static information about the network', async () => {
    const result = await client.query({
      query: await loadQueryNode('static')
    })
    expect(result.data).toMatchSnapshot()
  })
  it('Returns dynamic information about the network', async () => {
    const result = await client.query({
      query: await loadQueryNode('dynamic')
    })
    expect(result.data.cardano.tip.number).toBeGreaterThan(3994551)
    expect(result.data.cardano.currentEpoch.number).toBeGreaterThan(184)
  })
})
