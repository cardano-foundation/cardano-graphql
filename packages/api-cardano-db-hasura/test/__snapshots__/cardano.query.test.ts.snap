// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cardano Returns static information about the network 1`] = `
Object {
  "cardano": Object {
    "networkName": "mainnet",
    "protocolConst": 2160,
    "slotDuration": 20000,
    "startTime": "2017-09-23T21:44:51.000Z",
  },
}
`;
